src/coprocessor/local_metrics.rs

```diff
@@ -24,11 +24,11 @@ use pd::PdTask;
 /// `CopFlowStatistics` is for flow statistics, it would be reported to PD by flush.
 pub struct CopFlowStatistics {
     data: HashMap<u64, FlowStatistics>,
-    sender: FutureScheduler<PdTask>,
+    sender: Option<FutureScheduler<PdTask>>,
 }
 
 impl CopFlowStatistics {
-    pub fn new(sender: FutureScheduler<PdTask>) -> CopFlowStatistics {
+    pub fn new(sender: Option<FutureScheduler<PdTask>>) -> CopFlowStatistics {
```

Comment:
Seems `sender` will never be `None`?

Reply:
To align with the style with storage. In storage's `ReadPoolContext`, we use an optional FutureScheduler so that it is not required in tests. I kept this practice here.

Reply:
Similar to storage read pools, now in tests `sender` will be `None`.